### PR TITLE
Adds the BE for the new courses app...

### PR DESCRIPTION
**Pull request by kevinhowbrook**
This PR is the first part of the courses work. 
Designs: https://www.figma.com/file/H5pQECphbbYsvSWtag3OrCWC/Torchbox?type=design&node-id=3309-9169&mode=design&t=TxEdRwPd42H9EKdS-4

Note: This is part of a short 2 day brief that does not have any tickets created. FE will be done by Simon based of this.

### Description of Changes Made

In this PR, the following is added:
- New course detail page
- 2 new block types for the course detail page
- Stubbs out the course landing page model

**What's not been done:**
- The pattern library is currently broken and not rendering any pages. I have not fixed (or caused) this as part of this work

### How to Test

Checkout this branch locally.
Add a course landing page
Add a child course detail page.
You should be able to add everything needed for the course detail page, including the 2 new blocks.

### MR Checklist

- [ ] Add a description of your pull request and instructions for the reviewer to verify your work.
- [ ] If your pull request is for a specific ticket, link to it in the description.
- [ ] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Consider adding unit tests, especially for bug fixes. If you don't, tell us why.
- [ ] Tests and linting passes.
- [ ] Consider updating documentation. If you don't, tell us why.
- [ ] If relevant, list the environments / browsers in which you tested your changes.

### CODE REVIEW

File: tbx/courses/models.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        max_length=255, blank=True, help_text="e.g.£999 / $1,299 per person "
+    )
+    intro = wagtail_fields.RichTextField(blank=True, features=INTRO_RICHTEXT_FEATURES)
+    header_link = models.URLField(blank=True, help_text="e.g. https://www.example.com")
+    header_link_text = models.CharField(
+        max_length=255, blank=True, help_text="e.g. Visit example.com for dates"
+    )
+
+    body = wagtail_fields.StreamField(
+        tbx_course_blocks.CourseDetailStoryBlock(), use_json_field=True
```
Comment (William-Blackie) [inline tbx/courses/models.py:51]: Question: Do we still need `use_json_field` anymore?

File: tbx/courses/models.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        if self.header_link and not self.header_link_text:
+            errors["header_link_text"] = "You must set a text value for the link."
+
+        if self.header_link_text and not self.header_link:
+            errors["header_link"] = "You must set a link value for the header link."
+
+        if errors:
+            raise django_exceptions.ValidationError(errors)
+
+    def get_context(self, request):
```
Comment (William-Blackie) [inline tbx/courses/models.py:96]: This can go if we're not touching the context.

**Review by William-Blackie — APPROVED** [review]
Hey @kevinhowbrook, nice work, couple of small nitpicks.  :sparkles:

File: tbx/courses/models.py (hunk 2 above)
Comment (kevinhowbrook) [reply tbx/courses/models.py:96]: cheers, copypasta

File: tbx/courses/models.py (hunk 1 above)
Comment (kevinhowbrook) [reply tbx/courses/models.py:51]: :+1: not on 5.2 no

Comment (kevinhowbrook) [reply tbx/courses/models.py:51]: I think so https://docs.wagtail.org/en/stable/reference/streamfield/blocks.html#wagtail.fields.StreamField


> use_json_field – Must be set to True. This causes the field to use [JSONField](https://docs.djangoproject.com/en/stable/ref/models/fields/#django.db.models.JSONField) as its internal type, allowing the use of JSONField lookups and transforms.

**Pull request merged**